File: 1768-merge-strings-alternately/1768-merge-strings-alternately.c
char * mergeAlternately(char * word1, char * word2){
    char *result = (char *)malloc(sizeof(char) * (strlen(word1) + strlen(word2) + 1));
    int check = strlen(word1) < strlen(word2) ? strlen(word1) : strlen(word2);

    if (result == NULL) {
        return NULL;
    }
    
    int index = 0;

    for (int i = 0; i < check; i++) {
        result[index++] = word1[i];
        result[index++] = word2[i];
    }

    for (int i = check; word1[i] != '\0'; i++) {
        result[index++] = word1[i];
    }
    for (int i = check; word2[i] != '\0'; i++) {
        result[index++] = word2[i];
    }
    
    result[index] = '\0';

    return result;
}

File: 1768-merge-strings-alternately/1768-merge-strings-alternately.py
class Solution(object):
    def mergeAlternately(self, word1, word2):
        """
        :type word1: str
        :type word2: str
        :rtype: str
        """
        result = ""
        check = min(len(word1), len(word2))
        
        for i in range(check): 
            result += word1[i]
            result += word2[i]
        
        if len(word1) > check: 
            result += word1[i+1:]
        elif len(word2) > check: 
            result += word2[i+1:]
        
        return result